pkg/solidity-utils/test/foundry/Math.t.sol

```diff
@@ -11,19 +11,30 @@ contract MathTest is Test {
         uint256 abs = Math.abs(a);
 
         assertEq(abs, referenceAbs);
+        if (a >= 0) {
+            assertEq(int256(abs), a);
+        } else {
+            assertEq(int256(abs), -a);
+        }
     }
 
     function testMax(uint256 a, uint256 b) external {
-        uint256 referenceMax = (a < b) ? b : a;
         uint256 max = Math.max(a, b);
+        assertGe(max, a);
+        assertGe(max, b);
+        assertTrue((max == a) || (max == b));
 
+        uint256 referenceMax = (a < b) ? b : a;
         assertEq(max, referenceMax);
     }
```

Comment:
I would've split this into two tests - one comparing against the ref, and one testing props